EIPS/eip-draft-whisper.md

```diff
@@ -0,0 +1,138 @@
+## Preamble
+
+    EIP: draft
+    Title: Whisper Specification
+    Author: Vlad Gluhovsky <gluk256@gmail.com>
+    Type: Informational
+    Status: Draft
+    Created: 2017-05-05
+
+## Abstract
+
+This draft EIP describes the format of Whisper messages within the ÐΞVp2p Wire Protocol.
+This EIP should substitute the [existing specification](https://github.com/ethereum/wiki/wiki/Whisper-Wire-Protocol).
+More detailed documentation on Whisper could be found [here](https://github.com/ethereum/go-ethereum/wiki/Whisper).
+
+## Motivation
+
+It is necessary to specify the standard for Whipser messages in order to ensure forward compatibility of different Whisper clients.
+
+## Specification
+
+All Whisper messages sent as ÐΞVp2p Wire Protocol packets should be RLP-encoded arrays of data containing two objects: integer packet code followed by another object (whose type depends on the packet code). 
+
+If Whisper node does not support a particular packet code, it should just ignore the packet without generating any error.
+
+### Packet Codes
+
+The Whisper sub-protocol should support the following packet codes:
+
+| EIP   | Name                       | Int Value |
+|-------|----------------------------|-----------|
+|       | Status                     |   0x00    |
+|       | Messages                   |   0x01    |
+|       | P2P                        |   0x02    |
+|       | P2P Request                |   0x03    |
+|       | PoW Requirement            |   0x04    |
+|       | Bloom Filter               |   0x05    |
+
+
+### Packet Format and Usage
+
+**Status** [`0x00`, `whisper_protocol_version`] 
+
+This packet contains two objects: integer (0x00) followed by integer (Whisper protocol version).
+
+Informs a peer of the Whisper version. This message should be send after the initial handshake and prior to any other messages.
+
+**Messages** [`0x01`, `whisper_envelope`]
+
+This packet contains two objects: integer (0x01) followed by a single Whisper Envelope.
```

Comment:
Relaying only a single message per packet? It should be more efficient to allow batching.

Previous version of whisper also used the packet as a form of keep-alive: both peers on a connection would take turns sending `Messages` packets, starting with the peer with lower node key (comparing big-endian) and always responding within 1 second.